src/button-group/interfaces.ts

```diff
@@ -106,7 +121,12 @@ export interface InternalButtonGroupProps
     InternalBaseComponentProps {}
 
 export namespace ButtonGroupProps {
-  export type Variant = 'icon';
+  export type Variant =
+    | 'icon'
+    /**
+     * @awsuiSystem core
+     */
+    | 'children';
```

Comment:
I guess we do this as otherwise the following is confusing: `<ButtonGroup variant="icon" children={children} />`. However, this can be fixed if we allow both items and children to co-exist.